humilis_sam/resources.yaml.j2

```diff
@@ -1,11 +1,20 @@
 ---
 resources:
     {% if swagger_template %}
-    ApiGateway{{_layer.name}}{{_env.stage}}:
+    ApiKey:
+      Type: "AWS::ApiGateway::ApiKey"
+      DependsOn:
+        - ApiGateway
+      Properties:
+        Name: "{{_env.name}}-{{_env.stage|lower}}"
+        Description: "API key for stage {{_env.stage}} of environment {{_env.name}}"
+        Enabled: "true"
+
+    ApiGateway:
         Type: AWS::Serverless::Api
         Properties:
             DefinitionUri: {{swagger_template}}
-            StageName: {{_env.stage}}
+            StageName: {{_env.stage|lower}}
```

Comment:
With the way we currently use this layer, the stage won't appear in the URL (we include the stage name in the custom domain name, deploy one different API for each stage and attach one custom domain name by API : 1 stage - 1 API - 1 custom domain name containing the stage name). 
But if that changes in the future I agree having uppercase in the URL is ugly so let's fix it.